Add support for Comet Logging

sherpan opened the pull request:
# Related Issue or bug

None
Closes #[issue number that will be closed through this PR]

# Describe the changes you've made

Added support for logging to Comet 

# Type of change

Please delete options that are not relevant.
<!--
Example how to mark a checkbox :-
- [x] My code follows the code style of this project.
-->
- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Code style update (formatting, local variables)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [x] This change requires a documentation update

# How Has This Been Tested?

 `# load dataset
from pycaret.datasets import get_data
data = get_data('diabetes')

# init setup
from pycaret.classification import *
clf1 = setup(data, target = 'Class variable', log_experiment = 'comet_ml', experiment_name = 'asset_pycaret', log_data=True),

# model training
best_model = compare_models() `

# Describe if there is any unusual behaviour of your code(Write `NA` if there isn't)

N/A

# Checklist:

<!--
Example how to mark a checkbox :-
- [x] My code follows the code style of this project.
-->
- [x ] My code follows the style guidelines of this project.
- [x] I have performed a self-review of my own code.
- [x] I have commented my code, particularly in hard-to-understand areas.
- [x] I have made corresponding changes to the documentation.
- [x] My changes generate no new warnings.
- [x] I have added tests that prove my fix is effective or that my feature works.
- [x] New and existing unit tests pass locally with my changes.
- [x] Any dependent changes have been merged and published in downstream modules.

# Screenshots

 Original           | Updated
 :--------------------: |:--------------------:
 **original screenshot**  | <b>updated screenshot </b> |

ngupta23 commented on pycaret/nlp/__init__.py:
```diff
@@ -14,7 +14,12 @@
 from pycaret.loggers.base_logger import BaseLogger
 from pycaret.loggers.dagshub_logger import DagshubLogger
 from pycaret.loggers.mlflow_logger import MlflowLogger
+<<<<<<< HEAD
```
Looks like there is a conflict. Can you fix this?

sherpan replied:
should be fixed now!

celestinoxp commented on requirements-optional.txt:
```diff
@@ -31,6 +31,8 @@ fastapi>=0.75.0  # For web api
 m2cgen>=0.9.0  # For model conversion
 evidently>=0.1.45.dev0  # for drift reporting
+comet_ml>=3.31.22 #comet ml tracking
+joblib >=1.2.0 #file pipeliing
```
joblib exists in requirements.txt, i think you don't need insert insert in requirements-optional.txt

sherpan replied:
deleted!

moezali1 commented:
@sherpan Can you please also provide a colab notebook and demonstrate how the functionality is working?

Thanks.

sherpan commented:
@moezali1 

I put some example code in the PR but realized the formatting got messed up. Here it is. I can turn this into a colab when stuff is merged! 

```
from pycaret.datasets import get_data
data = get_data('diabetes')

# init setup
from pycaret.classification import *
clf1 = setup(data, target = 'Class variable', log_experiment = 'comet_ml', experiment_name = 'py_caret_diabetes', log_data=True),

# model training
best_model = compare_models() 
```

sherpan commented:
@celestinoxp @ngupta23  not sure why the code quality is failing. Im pretty sure I have installed the pre-commit hooks and didnt get anything. Any advice on how to re-produce the codestyle errors on my end so i can make sure its passing?

Yard1 commented:
@sherpan Can you run `pip install black==22.12.0 isort==5.11.4` and then run `black pycaret; isort --profile=black pycaret` manually?

celestinoxp commented:
> @sherpan Can you run `pip install black==22.12.0 isort==5.11.4` and then run `black pycaret; isort --profile=black pycaret` manually?

uma dúvida, porquê essas versões fixas? Não deveria instalar antes os requerimentos de requirements-dev.txt?

celestinoxp commented:
> @celestinoxp @ngupta23 not sure why the code quality is failing. Im pretty sure I have installed the pre-commit hooks and didnt get anything. Any advice on how to re-produce the codestyle errors on my end so i can make sure its passing?

If you use vscode, you might have to set black to format the code.
Just to confirm, make sure you have the https://github.com/pycaret/pycaret/blob/master/requirements-dev.txt dependencies installed and vscode (or other) configured to black format the code.
If this is the case with VSCode, I solved it with lines in settings.json:   
"editor.defaultFormatter": "ms-python.black-formatter",
"editor.formatOnType": true,
"editor.formatOnSave": true,

Yard1 commented:
@celestinoxp those are the versions we use in CI. I'll pin them in requirements as well.

sherpan commented:
thanks @celestinoxp @Yard1 for your instructions. I think i fixed it. here is what i got when i added my latest comet 
![Screen Shot 2023-02-02 at 4 12 29 PM](https://user-images.githubusercontent.com/16737376/216472235-72c671a4-168c-4580-85e6-070b1f383123.png)
